### PR TITLE
all: add Status messages to all statuses

### PR DESCRIPTION
**Pull request by carl-mastrangelo**

### CODE REVIEW

File: examples/src/main/java/io/grpc/examples/manualflowcontrol/ManualFlowControlServer.java [hunk 1]
```diff
@@ -101,7 +101,8 @@ public void onNext(HelloRequest request) {
               }
             } catch (Throwable throwable) {
               throwable.printStackTrace();
-              responseObserver.onError(Status.UNKNOWN.withCause(throwable).asException());
+              responseObserver.onError(
+                  Status.withDescription("Error handling request").UNKNOWN.withCause(throwable).asException());
```
Comment (ejona86) [inline examples/src/main/java/io/grpc/examples/manualflowcontrol/ManualFlowControlServer.java]: This one looks broken.

File: services/src/main/java/io/grpc/protobuf/services/ProtoReflectionService.java [hunk 2]
```diff
@@ -188,7 +188,10 @@ private void handleReflectionRequest() {
             listServices(request);
             break;
           default:
-            sendErrorResponse(request, Status.UNIMPLEMENTED, "");
+            sendErrorResponse(
+                request,
+                Status.Code.UNIMPLEMENTED,
+                "not implemented" + request.getMessageRequestCase());
```
Comment (ejona86) [inline services/src/main/java/io/grpc/protobuf/services/ProtoReflectionService.java]: Add a space before "

File: services/src/main/java/io/grpc/services/HealthServiceImpl.java [hunk 3]
```diff
@@ -41,7 +41,8 @@ public void check(HealthCheckRequest request,
       StreamObserver<HealthCheckResponse> responseObserver) {
     ServingStatus status = getStatus(request.getService());
     if (status == null) {
-      responseObserver.onError(new StatusException(Status.NOT_FOUND));
+      responseObserver.onError(new StatusException(
+          Status.NOT_FOUND.withDescription("unknown service" + request.getService())));
```
Comment (ejona86) [inline services/src/main/java/io/grpc/services/HealthServiceImpl.java]: Add a space before "

File: examples/src/main/java/io/grpc/examples/manualflowcontrol/ManualFlowControlServer.java (hunk 1 above)
Comment (carl-mastrangelo) [reply examples/src/main/java/io/grpc/examples/manualflowcontrol/ManualFlowControlServer.java]: Done

File: services/src/main/java/io/grpc/protobuf/services/ProtoReflectionService.java (hunk 2 above)
Comment (carl-mastrangelo) [reply services/src/main/java/io/grpc/protobuf/services/ProtoReflectionService.java]: Done

File: services/src/main/java/io/grpc/services/HealthServiceImpl.java (hunk 3 above)
Comment (carl-mastrangelo) [reply services/src/main/java/io/grpc/services/HealthServiceImpl.java]: Done

**Pull request merged**